File: app/src/main/java/com/numero/colorpalette/util/ColorUtil.java
package com.numero.colorpalette.util;

import android.graphics.Color;

public class ColorUtil {
    /**
     * return white or black color, depending on background color
     *
     * @param backgroundColor background color
     * @return Color.BLACK or Color.WHITE
     */
    public static int getTextColor(int backgroundColor) {
        int red = Color.red(backgroundColor);
        int green = Color.green(backgroundColor);
        int blue = Color.blue(backgroundColor);
        if ((red * 0.299 + green * 0.587 + blue * 0.114) > 186) {
            return Color.BLACK;
        } else {
            return Color.WHITE;
        }
    }
}

File: app/src/main/java/com/numero/colorpalette/util/ColorUtil.kt
package com.numero.colorpalette.util

import android.graphics.Color

object ColorUtil {
    /**
     * return white or black color, depending on background color
     *
     * @param backgroundColor background color
     * @return Color.BLACK or Color.WHITE
     */
    fun getTextColor(backgroundColor: Int): Int {
        val red = Color.red(backgroundColor)
        val green = Color.green(backgroundColor)
        val blue = Color.blue(backgroundColor)
        return if (red * 0.299 + green * 0.587 + blue * 0.114 > 186) {
            Color.BLACK
        } else {
            Color.WHITE
        }
    }
}
